text/metrics/0238-pipeline-monitoring.md

```diff
@@ -0,0 +1,426 @@
+definition above can be replaced for inclusive counters:
+
+```
+SingleLossRate = Total{success=false} / Total{*}
+```
+
+By using inclusive counters instead of exclusive counters, it is
+possible to establish the total rate of loss with substantially fewer
+timeseries, because we only need to distinguish success and failure
+the end of the pipeline to determine total loss.
+
+```
+PipelineLossRate = LastStageTotal{success=false} / FirstStageTotal{*}
+```
+
+Since total loss can be calculated with only a single timneseries per
+SDK, this will be specified as the behavior when configuring pipeline
+monitoring with basic-level metric detail.
+
+#### Collector perspective
+
+Collector counters are exclusive.  Like for SDKs, items that enter a
+processor are counted in one of three ways and to compute a meaningful
+ratio requires all three timeseries.  If the processor is a sampler,
+for example, the effective sampling rate is computed as
+`(accepted+refused)/(accepted+refused+dropped)`.
+
+While the collector defines and emits metrics sufficient for
+monitoring the individual pipeline component--taken as a whole, there
+is substantial redundancy in having so many exclusive counters.  For
+example, when a collector pipeline features no processors, the
+receiver's `refused` count is expected to equal the exporter's
+`send_failed` count.
+
+When there are several processors, it is primarily the number of
+dropped items that we are interested in counting.  When there are
+multiple sequential processors in a pipeline, however, counting the
+total number of items at each stage in a multi-processor pipeline
+leads to over-counting in aggregate.  For example, if you combine
+`accepted` and `refused` for two adjacent processors, then remove the
+metric attribute which distinguishes them, the resulting sum will be
+twice the number of items processed by the pipeline.
+
+### Pipeline monitoring
+
+The term _Stage_ is used to describe the a single component in an
+export pipeline.
+
+The term _Station_ is used to describe a location in the export
+pipeline where the participating stages are part of the same logical
+failure domain.  Typically each SDK or Collector is considered a
+station.
+
+#### Station integrity principles
+
+The [OpenTelemetry library guidelines (point
+4)](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/library-guidelines.md#requirements)
+describes a separation of protocol-dependent ("receivers",
+"exporters") and protocol-independent ("processors") parts.  We refer
+to the combination of parts as a station.
+
+The station concept is called out because within a station, we expect
+that the station (software) acts responsibly by design, for the
+integrity of the pipeline.  Stations allow data to enter a pipeline
+only through receiver components.  Stations are never responsible for
+dropping data, because only processor components drop data.  Stations
```

Comment:
SDK processors do not drop data, and processors aren't responsible for passing data to the next process in a pipeline. Instead, higher order SDK code ensures that each registered processor is called. As the [spec puts it](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#span-processor), "Each processor registered on TracerProvider is a start of pipeline that consist of span processor and optional exporter".

The architectural differences between SDK and collector processor may impact your design. For example, further on you state:

> 1. Data that enters a station is eventually exported or dropped.
2. No other outcomes are possible.

For SDKs, its perfect valid for a processor to do nothing besides extract baggage and add it to the span. No filtering, no exporting.